hummingbot/connector/exchange/wazirx/wazirx_in_flight_order.py

```diff
@@ -0,0 +1,99 @@
+from decimal import Decimal
+from typing import (
+    Any,
+    Dict,
+    Optional,
+)
+import asyncio
+from hummingbot.core.event.events import (
+    OrderType,
+    TradeType
+)
+from hummingbot.connector.in_flight_order_base import InFlightOrderBase
+
+
+class WazirxInFlightOrder(InFlightOrderBase):
+    def __init__(self,
+                 client_order_id: str,
+                 exchange_order_id: Optional[str],
+                 trading_pair: str,
+                 order_type: OrderType,
+                 trade_type: TradeType,
+                 price: Decimal,
+                 amount: Decimal,
+                 initial_state: str = "OPEN"):
+        super().__init__(
+            client_order_id,
+            exchange_order_id,
+            trading_pair,
+            order_type,
+            trade_type,
+            price,
+            amount,
+            initial_state,
+        )
+        self.trade_id_set = set()
+        self.cancelled_event = asyncio.Event()
+
+    @property
+    def is_done(self) -> bool:
+        return self.last_state in {"cancel", "done"}
+
+    @property
+    def is_failure(self) -> bool:
+        return self.last_state in {"cancel"}
+
+    @property
+    def is_cancelled(self) -> bool:
+        return self.last_state in {"cancel"}

```

Comment:
Failure and cancelled should not be mapped to the same status. Cancellation is something expected. Failure is cause because of an error.

Reply:
Updated, made is_failure() as "fail". But this is not defined in api.